client/scripts/startup.js

```diff
@@ -1,13 +1,24 @@
 /* eslint-disable global-require */
-function start() {
-  require('../app/bundles/Course/LessonPlan/startup');
+/* eslint-disable import/no-dynamic-require */
+
+function loadCurrentModule() {
+  const modulePath = document.head.querySelector("[name='server-context']")
+                       .getAttribute('data-controller-name');
+
+  try {
+    require(`../app/bundles/${modulePath}`);
+  } catch (e) {
+    // Module does not exist
+  }
 }
 
 if (!global.Intl) {
-  require('intl');
-  require('intl/locale-data/jsonp/en');
-  require('intl/locale-data/jsonp/zh');
-  start();
+  require.ensure([], (require) => {
+    require('intl');
+    require('intl/locale-data/jsonp/en');
+    require('intl/locale-data/jsonp/zh');
+    loadCurrentModule();
+  }, 'intl');
 } else {
-  start();
+  loadCurrentModule();
```

Comment:
Why not bring `loadCurrentModule()` out of both blocks? 

Will we have cases of other packages like `intl`, where the dependency cannot be inferred by the `server_meta_tag`?